clients/vscode-hlasmplugin/src/test/suite/hlasmListingServices.test.ts

```diff
@@ -0,0 +1,236 @@
+    let services: ReturnType<typeof createListingServices>;
+    let document: vscode.TextDocument;
+    let listings: ReturnType<ReturnType<typeof createListingServices>['handleListingContent']>;
+
+    suiteSetup(async () => {
+        services = createListingServices();
+        document = await vscode.workspace.openTextDocument({ language: 'hlasmListing', content: sampleListing });
+        listings = services.handleListingContent(document);
+        assert(listings);
+    })
+
+    suiteTeardown(async () => {
+        services.releaseListingContent(document);
+    })
+
+    test('Validate listing structure', async () => {
+        assert.strictEqual(listings?.length, 1);
+
+        const l = listings[0];
+
+        assert.deepStrictEqual(l.options, { start: 2, end: 69 });
+        assert.deepStrictEqual(l.externals, { start: 69, end: 73 });
+        assert.deepStrictEqual(l.codeSections, [{ start: 76, end: 86, title: '' }]);
+        assert.strictEqual(l.relocations, undefined);
+        assert.deepStrictEqual(l.ordinary, { start: 86, end: 93 });
+        assert.strictEqual(l.macro, undefined);
+        assert.deepStrictEqual(l.dsects, { start: 93, end: 97 });
+        assert.deepStrictEqual(l.usings, { start: 97, end: 103 });
+        assert.deepStrictEqual(l.registers, { start: 103, end: 122 });
+        assert.deepStrictEqual(l.summary, { start: 122, end: 142 });
+
+    })
+
+    test('Definition in code block', async () => {
+        const def1 = services.provideDefinition(document, new vscode.Position(79, 110));
+        const def2 = services.provideDefinition(document, new vscode.Position(80, 56));
+
+        assert.deepStrictEqual(def1, def2);
+        assert.strictEqual(def1?.length, 1);
+        assert.deepStrictEqual(def1[0].range.start.line, 83);
+    });
+
+    test('Definition outside of code block', async () => {
+        const def = services.provideDefinition(document, new vscode.Position(101, 90));
+
+        assert.strictEqual(def?.length, 1);
+        assert.deepStrictEqual(def[0].range.start.line, 81);
+    });
+
+    test('References', async () => {
+        const ref = services.provideReferences(document, new vscode.Position(101, 90), { includeDeclaration: false });
+
+        assert.strictEqual(ref?.length, 1);
+        assert.deepStrictEqual(ref[0].range.start.line, 77);
+    });
+
+    test('Hover', async () => {
+        const h = services.provideHover(document, new vscode.Position(101, 90));
+
+        const md = h?.contents[0];
+        assert.ok(md instanceof vscode.MarkdownString);
+        assert.ok(md.value.includes('DSECT'));
+    });
+
+    test('Outline', async () => {
+        const outline = services.provideDocumentSymbols(document);
+
+        assert.strictEqual(outline?.length, 1);
+        const code = outline[0].children.find(x => x.children.length > 0);
+        assert.ok(code);
+        assert.ok(code.children.some(x => x.name === '(untitled)'));
+    });
```

Comment:
Ok, I can get several listings in one file (e.g. by copy pasting) and then this would be useful.

Nevertheless, maybe the first listing in the outline should be just `Listing` since the usual file will not contain multiple listings?